workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityComponentDataGenerator.tt

```diff
@@ -213,6 +193,16 @@ var fieldDetails = fieldDetailsList[i]; #>
 <# } #>
             }
 
+            public static void SerializeSnapshot(<#= componentNamespace #>.Snapshot snapshot, global::Improbable.Worker.CInterop.SchemaObject obj)
+            {
+<# for (var i = 0; i < fieldDetailsList.Count; i++) {
+var fieldDetails = fieldDetailsList[i]; #>
+                {
```

Comment:
scoping isn't needed here

Reply:
It is as the serialisation string may be multi-line with local variables